.github/workflows/benchmark.yml

```diff
@@ -0,0 +1,81 @@
+name: Performance Benchmark
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
+
+jobs:
+  benchmark:
+    name: Run Performance Benchmarks
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0  # Get full history for comparison
+
+      - name: Set up Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version: '20'
+          cache: 'npm'
+
+      - name: Install dependencies
+        run: npm ci
+
+      # Install benchmark dependencies
+      - name: Install benchmark dependencies
+        run: cd benchmark && npm ci
+
+      # For PRs: Run benchmarks on both the base branch and the PR branch
+      - name: Compare with base branch
+        if: ${{ github.event_name == 'pull_request' }}
+        run: |
+          # Save current branch name to go back later
+          CURRENT_BRANCH=$(git rev-parse --abbrev-ref HEAD)
+
+          # Create directories for results
+          mkdir -p /tmp/qs-benchmark/base
+          mkdir -p /tmp/qs-benchmark/pr
+
+          echo "==== Running benchmarks on base branch ===="
+          # Check out the base branch
+          git checkout origin/${{ github.base_ref }}
+          # Install dependencies (in case they changed)
+          npm ci
+          cd benchmark && npm ci && cd ..
```

Comment:
please use `npm install` throughout; packages shouldn't have lockfiles and this one doesn't.